From 6814f2f6248d22b1fdf78b46ad0a7f97a7087a65 Mon Sep 17 00:00:00 2001
From: Jason Wang <jason.ruonan.wang@gmail.com>
Date: Sat, 2 Apr 2022 17:12:40 -0400
Subject: [PATCH 1/2] only build dataspaces when MPI is found

---
 cmake/DetectOptions.cmake | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/cmake/DetectOptions.cmake b/cmake/DetectOptions.cmake
index aa5159e6e..d4e2e5dbf 100644
--- a/cmake/DetectOptions.cmake
+++ b/cmake/DetectOptions.cmake
@@ -255,13 +255,15 @@ elseif(ADIOS2_USE_MHS)
 endif()
 
 # DataSpaces
-if(ADIOS2_USE_DataSpaces STREQUAL AUTO)
-  find_package(DataSpaces 2.1.1)
-elseif(ADIOS2_USE_DataSpaces)
-  find_package(DataSpaces 2.1.1 REQUIRED)
-endif()
-if(DATASPACES_FOUND)
-  set(ADIOS2_HAVE_DataSpaces TRUE)
+if(MPI_FOUND)
+    if(ADIOS2_USE_DataSpaces STREQUAL AUTO)
+        find_package(DataSpaces 2.1.1)
+    elseif(ADIOS2_USE_DataSpaces)
+        find_package(DataSpaces 2.1.1 REQUIRED)
+    endif()
+    if(DATASPACES_FOUND)
+        set(ADIOS2_HAVE_DataSpaces TRUE)
+    endif()
 endif()
 
 # HDF5

From 63e5d6c0e3f3cb78350bc87f3193de88241c3b26 Mon Sep 17 00:00:00 2001
From: Jason Wang <jason.ruonan.wang@gmail.com>
Date: Sun, 3 Apr 2022 21:16:00 -0400
Subject: [PATCH 2/2] fix template instantiation bug in DataMan serializer

---
 .../adios2/toolkit/format/dataman/DataManSerializer.cpp   | 1 -
 source/adios2/toolkit/format/dataman/DataManSerializer.h  | 8 ++++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/source/adios2/toolkit/format/dataman/DataManSerializer.cpp b/source/adios2/toolkit/format/dataman/DataManSerializer.cpp
index e343fb76f..e6e20fdb1 100644
--- a/source/adios2/toolkit/format/dataman/DataManSerializer.cpp
+++ b/source/adios2/toolkit/format/dataman/DataManSerializer.cpp
@@ -622,7 +622,6 @@ void DataManSerializer::Log(const int level, const std::string &message,
     }
 }
 
-template <>
 void DataManSerializer::PutData(
     const std::string *inputData, const std::string &varName,
     const Dims &varShape, const Dims &varStart, const Dims &varCount,
diff --git a/source/adios2/toolkit/format/dataman/DataManSerializer.h b/source/adios2/toolkit/format/dataman/DataManSerializer.h
index 65e4b430a..44647ccbd 100644
--- a/source/adios2/toolkit/format/dataman/DataManSerializer.h
+++ b/source/adios2/toolkit/format/dataman/DataManSerializer.h
@@ -93,6 +93,14 @@ class DataManSerializer
     void PutAttributes(core::IO &io);
 
     // put a variable for writer
+    void PutData(const std::string *inputData, const std::string &varName,
+                 const Dims &varShape, const Dims &varStart,
+                 const Dims &varCount, const Dims &varMemStart,
+                 const Dims &varMemCount, const std::string &doid,
+                 const size_t step, const int rank, const std::string &address,
+                 const std::vector<std::shared_ptr<core::Operator>> &ops,
+                 VecPtr localBuffer = nullptr, JsonPtr metadataJson = nullptr);
+
     template <class T>
     void PutData(const T *inputData, const std::string &varName,
                  const Dims &varShape, const Dims &varStart,
