diff --git a/cmake/DetectOptions.cmake b/cmake/DetectOptions.cmake
index f6eacb178..fc818e4f0 100644
--- a/cmake/DetectOptions.cmake
+++ b/cmake/DetectOptions.cmake
@@ -509,12 +509,13 @@ if(Python_Interpreter_FOUND)
     "${_Python_DEVPREFIX}" "${Python_SITEARCH}" CMAKE_INSTALL_PYTHONDIR_DEFAULT
   )
   set(CMAKE_INSTALL_PYTHONDIR "${CMAKE_INSTALL_PYTHONDIR_DEFAULT}"
-    CACHE PATH "Install directory for python modules"
+    CACHE STRING "Install directory for python modules"
   )
   mark_as_advanced(CMAKE_INSTALL_PYTHONDIR)
   set(CMAKE_PYTHON_OUTPUT_DIRECTORY
     ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_PYTHONDIR}
   )
+  message(STATUS "DEBUG: CMAKE_INSTALL_PYTHONDIR=${CMAKE_INSTALL_PYTHONDIR} CMAKE_PYTHON_OUTPUT_DIRECTORY=${CMAKE_PYTHON_OUTPUT_DIRECTORY}")
 endif()
 
 # Sst
